### PR TITLE
Minor javadoc updates around unpick

### PR DESCRIPTION
**Pull request by liach**
A few things:
1. Added licenser and cc0 message to constant classes
2. Add constant classes to jd gen, wrote a package info that partially addressed #2195 
3. Fixed one javadoc error in current yarn
4. Changed set block state method doc to use the doc of the constant holding class instead

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -12,4 +12,4 @@ jobs:
     steps:
       - uses: actions/checkout@v1
       - uses: gradle/wrapper-validation-action@v1
-      - run: ./gradlew build javadocJar checkMappings mapNamedJar --stacktrace
+      - run: ./gradlew checkLicenses build javadocJar checkMappings --stacktrace
```
Comment (modmuss50) [inline .github/workflows/build.yml]: No, why remove `mapNamedJar` ?

**Review by modmuss50 — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml (hunk 1 above)
Comment (liach) [reply .github/workflows/build.yml]: It is a dependency of `genFakeSource`, which leads to `javadoc` and `javadocJar`

Comment (liach) [reply .github/workflows/build.yml]: The proof that this is ran: https://github.com/FabricMC/yarn/runs/2212467447#step:5:181

Comment (modmuss50) [reply .github/workflows/build.yml]: Ok cool, all good then 👍

**Review by modmuss50 — APPROVED** [review]

**Review by Earthcomputer — APPROVED** [review]

File: .github/workflows/build.yml (hunk 1 above)
Comment (Juuxel) [inline .github/workflows/build.yml]: Hmm isn't `checkLicenses` a dependency of `check` (and thus `build`)?

File: mappings/net/minecraft/block/entity/AbstractFurnaceBlockEntity.mapping [hunk 2]
```diff
@@ -50,7 +50,8 @@ CLASS net/minecraft/class_2609 net/minecraft/block/entity/AbstractFurnaceBlockEn
 	METHOD method_17763 dropExperienceForRecipesUsed (Lnet/minecraft/class_3222;)V
 		ARG 1 player
 	METHOD method_26395 isNonFlammableWood (Lnet/minecraft/class_1792;)Z
-		COMMENT Returns whether the provided {@code item} is in the {@linkplain ItemTags#NON_FLAMMABLE_WOOD} tag.
+		COMMENT Returns whether the provided {@code item} is in the {@linkplain
+		COMMENT net.minecraft.tag.ItemTags#NON_FLAMMABLE_WOOD non_flammable_wood} tag.
```
Comment (Juuxel) [inline mappings/net/minecraft/block/entity/AbstractFurnaceBlockEntity.mapping]:
```suggestion
		COMMENT Returns whether the provided {@code item} is in the {@link
		COMMENT net.minecraft.tag.ItemTags#NON_FLAMMABLE_WOOD non_flammable_wood} tag.
```
I'd say a non-plaintext link is better here.

**Review by Juuxel — APPROVED** [review]

File: .github/workflows/build.yml (hunk 1 above)
Comment (liach) [reply .github/workflows/build.yml]:
```suggestion
      - run: ./gradlew build javadocJar checkMappings --stacktrace
```
It indeed is according to gradle build scan. Optimizing.

File: mappings/net/minecraft/block/entity/AbstractFurnaceBlockEntity.mapping (hunk 2 above)
Comment (liach) [reply mappings/net/minecraft/block/entity/AbstractFurnaceBlockEntity.mapping]: Hmm, seems link is like wrapping stuff in `<code>`, so this actually works better I guess?

**Review by YanisBft — APPROVED** [review]

File: mappings/net/minecraft/block/entity/AbstractFurnaceBlockEntity.mapping (hunk 2 above)
Comment (Juuxel) [reply mappings/net/minecraft/block/entity/AbstractFurnaceBlockEntity.mapping]: Yeah `{@link}` makes it use the code font which is better for a tag.

**Comment by liach** [comment]
Additionally, after merging some unpick prs, we need to run `gradle licenseFormat` in order to add the cc0 license to our constant classes (arr bad)

**Pull request merged**